planner/core/preprocess.go

```diff
@@ -101,14 +101,11 @@ func Preprocess(ctx sessionctx.Context, node ast.Node, preprocessOpt ...Preproce
 		v.PreprocessorReturn = &PreprocessorReturn{}
 	}
 	node.Accept(&v)
-	readTS := ctx.GetSessionVars().TxnReadTS.UseTxnReadTS()
```

Comment:
Moved to `handleAsOfAndReadTS` to center the initialize of `LastSnapshotTS` and other parameters.